### ISSUE TITLE
[docs] Improve documentation about updatable manifest properties through OTA

### ISSUE THREAD

**Issue by Aryk** [expo/expo#11269]
I was looking through the Expo documentation and couldn’t find a definitive list of which keys in app.json can be changed without requiring a re-build and re-submit to the app store. Does this list exist somewhere? 

I think this should be in the documentation to avoid pointless wasting of time trying to deduce it by yourself.
 
I've skimmed through the forums as well and you have other people who also "aren't quite sure"....

https://forums.expo.io/t/expo-version-number-not-updated-on-ota-updates/29164/2 
https://forums.expo.io/t/what-app-json-properties-can-be-delivered-ota/9151/2

Maybe we can add it for each item here: 

https://docs.expo.io/versions/latest/config/app/

**Comment by byCedric** [comment]
Hi @Aryk! That's a valid point, although almost everything should be not-OTA-updatable. This is mostly because all properties affecting native configuration requires a new build to fully propagate them. Take the `android.permissions` for example, this affects the `<use-permission>` tags in the standalone build. When you add or remove some of these, they need to be propagated to native code for Android to pick it up.

We do have configuration that won't affect native code, e.g.
- the [`description`](https://docs.expo.io/versions/latest/config/app/#description) or [`githubUrl`](https://docs.expo.io/versions/latest/config/app/#githuburl) property, only affects how your app is displayed on expo.io
- the [`extra`](https://docs.expo.io/versions/latest/config/app/#extra) property, only contains custom data provided by dev and thus not affecting native code

[This doc briefly talks about what isn't updatable through OTA](https://docs.expo.io/workflow/publishing/#some-native-configuration-cant-be-updated-by). Maybe that might help you for now.

**Comment by Aryk** [comment]
Thanks for pointing me to that, I must have skipped over it, but it doesn't seem conclusive and 100% accurate. It says anything under ios and android requires rebuild "for changes to take effect", yet...

I added `playStoreUrl` and `appStoreUrl` and published it (without rebuilding the binaries), and found I was able to access the values through the Constants.manifest.platform.

So that worked, but then maybe `expo-store-review` will not work unless I rebuild?

That's the thing, you guys say it won't work but some (or maybe all) the functionality does which makes me discount the documentation as more of a "suggestion" then the accurate thing. Or did I find the **one** exception?
